pkg/actions/iam/update.go

```diff
@@ -0,0 +1,62 @@
+package iam
+
+import (
+	"fmt"
+
+	"github.com/weaveworks/eksctl/pkg/ctl/cmdutils"
+
+	"github.com/weaveworks/eksctl/pkg/cfn/manager"
+
+	"github.com/kris-nova/logger"
+
+	"github.com/weaveworks/eksctl/pkg/utils/tasks"
+
+	api "github.com/weaveworks/eksctl/pkg/apis/eksctl.io/v1alpha5"
+)
+
+func (a *Manager) UpdateIAMServiceAccounts(iamServiceAccounts []*api.ClusterIAMServiceAccount, plan bool) error {
+	var nonExistingSAs []string
+	updateTasks := &tasks.TaskTree{Parallel: true}
+
+	stacks, err := a.stackManager.ListStacksMatching("eksctl-.*-addon-iamserviceaccount")
+	if err != nil {
+		return err
+	}
+
+	existingStacksMap := listToMap(stacks)
+
+	for _, iamServiceAccount := range iamServiceAccounts {
+		stackName := makeIAMServiceAccountStackName(a.clusterName, iamServiceAccount.Namespace, iamServiceAccount.Name)
+
+		if _, ok := existingStacksMap[stackName]; !ok {
+			logger.Info("Cannot update IAMServiceAccount %s/%s as it does not exist", iamServiceAccount.Namespace, iamServiceAccount.Name)
+			nonExistingSAs = append(nonExistingSAs, fmt.Sprintf("%s/%s", iamServiceAccount.Namespace, iamServiceAccount.Name))
+			continue
+		}
+
+		taskTree, err := NewUpdateIAMServiceAccountTask(a.clusterName, iamServiceAccount, a.stackManager, iamServiceAccount, a.oidcManager)
+		if err != nil {
+			return err
+		}
+		taskTree.PlanMode = plan
+		updateTasks.Append(taskTree)
+	}
+	if len(nonExistingSAs) > 0 {
+		logger.Info("the following IAMServiceAccounts will not be updated as they do not exist: %v", nonExistingSAs)
+	}
+
+	defer cmdutils.LogPlanModeWarning(plan && len(iamServiceAccounts) > 0)
+	return doTasks(updateTasks)
+
+}
+
+func listToMap(stacks []*manager.Stack) map[string]string {
+	stacksMap := make(map[string]string)
+	for _, stack := range stacks {
+		stacksMap[*stack.StackName] = ""
```

Comment:
kind of nitpicky sorry but i think this is a set, right? I think it's more idiomatic to use `struct{}` as the value type then.

Reply:
*technically* it is a map 😉 , but yeah its used as a set. I'll refactor the name and use`struct{}`